chia/daemon/server.py

```diff
@@ -604,6 +608,39 @@ async def _state_changed(self, service: str, message: Dict[str, Any]):
     def state_changed(self, service: str, message: Dict[str, Any]):
         asyncio.create_task(self._state_changed(service, message))
 
+    async def _watch_log_file(self, fp: TextIO, loop: asyncio.AbstractEventLoop):
+        """
+        Loops every 5 seconds and reads in data from the log file and sends out update
+        Stops when there are no more websockets open for this service
+        """
+        await asyncio.sleep(5)
+        while True:
+
+            # Get all websockets opened for this particular service
+            websockets = self.connections.get(service_logs)
+            if websockets is None or not websockets:
+                fp.close()
```

Comment:
If you use the `with open` idom to open fp, you can remove `fp.close()` here.

Best to keep the open and close together.